StortSpelProjekt/Engine/src/ECS/Components/GUI2DComponent.h

```diff
@@ -0,0 +1,78 @@
+#ifndef GUI2DCOMPONENT_H
+#define GUI2DCOMPONENT_H
+
+#include <map>
+#include <vector>
+#include "Component.h"
+class Window;
+class Texture;
+class Text;
+class Font;
+class Mesh;
+class Resource;
+class ShaderResourceView;
+class Events;
+class Eventbus;
+
+struct TextData;
+struct Vertex;
+struct ButtonPressed;
+
+namespace component
+{
+	class GUI2DComponent : public Component
+	{
+	public:
+		GUI2DComponent(Entity* parent);
+		virtual ~GUI2DComponent();
+
+		// Text functions
+		std::map<std::string, TextData>* const GetTextDataMap();
+		TextData* GetTextData(std::string name);
+
+		void AddText(std::string name);
+		void SubmitText(Text* text, std::string name);
+		void ReplaceText(Text* text, std::string name);
+
+		void SetFont(Font* font);
+		void SetText(std::string text, std::string name);
+		void SetPos(float2 textPos, std::string name);
+		void SetScale(float2 scale, std::string name);
+		void SetPadding(float2 padding, std::string name);
+		void SetColor(float4 color, std::string name);
+
+		Font* GetFont() const;
+		Texture* GetFontTexture() const;
+		Text* GetText(std::string name);
+		std::map<std::string, Text*>* GetTextMap();
+		const int GetNumOfTexts() const;
+		const int GetNumOfCharacters(std::string name);
+
+		void UploadTextData(std::string name);
```

Comment:
At the moment, 2dGuiComponent got alot of public functions, making it abit hard for the eye to manage. Is it possible to make it abit more clear by (for example)  splitting the text & quad part into two separate classes which the 2dGuiComponent then uses? In that case, all the developer sees in main is something like: 2dGuiComponent->GetTextmanager->setText. Just a thought

Reply:
Yeah sure, I will make two manager classes